src/data/real/hyperreal.lean

```diff
@@ -250,13 +250,25 @@ lemma is_st_iff_abs_sub_lt_delta {x : ℝ*} {r : ℝ} :
 by simp only [abs_sub_lt_iff, @sub_lt _ _ (r : ℝ*) x _,
     @sub_lt_iff_lt_add' _ _ x (r : ℝ*) _, and_comm]; refl
 
+section norm_cast_workaround
+
+local attribute [simp] private lemma two : (2 : ℝ*) = 1 + 1 := rfl
+local attribute [simp] private lemma two' : (2 : ℝ) = 1 + 1 := rfl
+
+-- Otherwise the following fails:
+-- #eval norm_cast.coe_to_numeral `((2 : ℕ) : ℝ*)
+-- #eval norm_cast.coe_to_numeral `((2 : ℕ) : ℝ)
+-- #eval norm_cast.numeral_to_coe `(2 : ℝ)
```

Comment:
This is because `nat.cast_succ` applies to `bit0 has_one.one` and hence simplifies `↑2` to `↑1 + 1` and then `1 + 1` instead of `bit0 1`.

Reply:
I'm not sure why this happens.  Both `nat.cast_succ` and `nat.cast_bit0` are tagged `@[simp, norm_cast]` and `nat.cast_bit0` is defined after `nat.cast_succ` so it should be used preferentially (and is used preferentially by simp).

MWE:
```lean
import data.nat.cast tactic.norm_cast

constant ℝ : Type
@[instance] constant real.add_monoid : add_monoid ℝ
@[instance] constant real.has_one : has_one ℝ

set_option trace.simplify.rewrite true
set_option pp.notation false
set_option pp.numerals false

-- uses nat.cast_bit0
example : ((2 : ℕ) : ℝ) = 2 := by simp

-- fails
#eval norm_cast.numeral_to_coe `(2 : ℝ)
#eval norm_cast.coe_to_numeral `((2 : ℕ) : ℝ)
```

Reply:
I think this is due to a quirk in how user attribute calls `mk_cache`: the argument with the list of declarations is in reverse order.

Reply:
Indeed, that fixed it.  I've also added a test for the issue.